### ISSUE TITLE
yogakit ,The program crashes when I assign a yoga's  value to other view's yoga

### ISSUE THREAD

**Issue by D8Ge** [facebook/yoga#684]
# Report

- [x] I have searched [existing issues](https://github.com/facebook/yoga/issues) and this is not a duplicate

# Issues and Steps to Reproduce
I want to reuse the layout, so i had do that 
![2017-12-10 9 59 37](https://user-images.githubusercontent.com/19723961/33801326-36299bf0-dd93-11e7-9bf9-f8d1625a798b.png)
and call it:
![2017-12-10 11 22 29](https://user-images.githubusercontent.com/19723961/33801654-a513e026-dd9c-11e7-9892-bc9b51c5b068.png)

but i got a error:
![2017-12-10 9 54 15](https://user-images.githubusercontent.com/19723961/33801334-753113f0-dd93-11e7-8f70-448a663185c4.png)

# Expected Behavior
When I use yogakit, I have a lot of duplication of code , so i want to reuse the layout.

# Actual Behavior
when i assign value to a view'yoga. there are some mistake

# Link to Code
[https://github.com/D8Ge/ReuseYoga/blob/master/ReuseYoga.swift](url)

***When applicable, use this [fiddle](https://jsfiddle.net/emilsjolander/jckmwztt/) to post a web repro.***

**Comment by priteshrnandgaonkar** [comment]
The error happens when you set a property on a layout.
```
   let layout = YGLayout()
   layout.isEnabled = true
   layout.flexDirection = .row
   layout.justifyContent = .center
   return layout
```
This doesn't `let layout = YGLayout()` initialise the internal `YGNode` class. You are not meant to initialise `YGLayout` on your own. Internally `YGLayout` is initialised with different api which includes `view` in its argument. In this initialiser `YGNode` is initialised which is mandatory for the setters and getters of `YGLayout` class to work properly.

TL;DR: You are not meant to initialise `YGLayout` on your own.

**Comment by priteshrnandgaonkar** [comment]
We should hide the default `init()` method

**Comment by priteshrnandgaonkar** [comment]
Thanks for the issue.The default `init` method is made unavailable here https://github.com/facebook/yoga/commit/dbf6a121348ed81c24f8d48244cb390e553894a4 , so now one can't do `let layout = YGLayout()`.
